TWTValidation/JSON Validator/Schema Validators/TWTJSONSchemaObjectValidator.m

```diff
@@ -0,0 +1,185 @@
+                          propertyValidators:(NSArray *)propertyValidators
+                   patternPropertyValidators:(NSArray *)patternPropertyValidators
+               additionalPropertiesValidator:(TWTValidator *)additionalPropertiesValidator
+                        propertyDependencies:(NSDictionary *)propertyDependencies
+{
+    self = [super init];
+    if (self) {
+        _maximumPropertyCount = maximumPropertyCount;
+        _minimumPropertyCount = minimumPropertyCount;
+        _requiredPropertyKeys = [requiredPropertyKeys copy];
+        _propertyValidators = [propertyValidators copy];
+        _patternPropertyValidators = [patternPropertyValidators copy];
+        _additionalPropertiesValidator = [additionalPropertiesValidator copy];
+        _propertyDependencies = [propertyDependencies copy];
+
+        //  Make property validators accessible by their key
+        NSMutableDictionary *propertyValidatorsByKey = [[NSMutableDictionary alloc] init];
+        for (TWTKeyValuePairValidator *pairValidator in propertyValidators) {
+            [propertyValidatorsByKey setObject:pairValidator forKey:pairValidator.key];
+        }
+        _propertyAndValidators = [propertyValidatorsByKey copy];
+
+    }
+    return self;
+}
+
+
+- (instancetype)init
+{
+    return [self initWithMaximumPropertyCount:nil minimumPropertyCount:nil requiredPropertyKeys:nil propertyValidators:nil patternPropertyValidators:nil
+                additionalPropertiesValidator:nil propertyDependencies:nil];
+}
+
+
+- (BOOL)validateValue:(id)value error:(out NSError *__autoreleasing *)outError
+{
+    if (![super validateValue:value error:outError]) {
+        return NO;
+    } else if (![value isKindOfClass:[NSDictionary class]]) {
+        if (outError) {
+            *outError = [NSError twt_validationErrorWithCode:TWTValidationErrorCodeValueHasIncorrectClass
+                                            failingValidator:self
+                                                       value:value
+                                        localizedDescription:TWTLocalizedString(@"TWTJSONSchemaObjectValidator.notDictionaryError")];
+        }
+        return NO;
+    }
+
+    BOOL countValidated = YES;
+    BOOL requiredPropertiesValidated = YES;
+    BOOL propertiesValidated = YES;
+    BOOL patternPropertiesValidated = YES;
+    BOOL additionalPropertiesValidated = YES;
+    BOOL dependenciesValidated = YES;
+
+    NSError *countError = nil;
+    NSError *requiredPropertiesError = nil;
+    NSMutableArray *propertiesErrors = outError ? [[NSMutableArray alloc] init] : nil;
+    NSMutableArray *patternPropertiesErrors = outError ? [[NSMutableArray alloc] init] : nil;
+    NSMutableArray *additionalPropertiesErrors = outError ? [[NSMutableArray alloc] init] : nil;
+    NSMutableArray *dependenciesErrors = outError ? [[NSMutableArray alloc] init] : nil;
+
+    NSSet *keysSet = [NSSet setWithArray:[value allKeys]];
```

Comment:
Both sound off. I thought about "allKeys" but wasn't sure if it'd be confusing with value.allKeys being an array. What about just "keys"?